drink/src/lib.rs

```diff
@@ -44,13 +45,19 @@ pub type EventRecordOf<Runtime> = EventRecord<
     <Runtime as frame_system::Config>::Hash,
 >;
 
+type ContractBalanceOf<R> =
+    <<R as pallet_contracts::Config>::Currency as Inspect<AccountIdFor<R>>>::Balance;
```

Comment:
Ah, I see we have different types for "Balances" and "ContractBalances". This, again _just_, feels wrong - but I understand this comes from Substrate deps. I think it's a place where a _serious_ mistakes can happen since some tools (like ContractsUI for example) have a special handling for `Balance` type.